runners/flink/src/main/java/org/apache/beam/runners/flink/LyftFlinkStreamingPortableTranslations.java

```diff
@@ -357,6 +377,65 @@ private void translateKinesisInput(
             .addSource(source, FlinkLyftKinesisConsumer.class.getSimpleName()));
   }
 
+  private void translateS3AndKinesisInputs(
+      String id,
+      RunnerApi.Pipeline pipeline,
+      FlinkStreamingPortablePipelineTranslator.StreamingTranslationContext context) {
+    RunnerApi.PTransform pTransform = pipeline.getComponents().getTransformsOrThrow(id);
+
+    ObjectMapper mapper = new ObjectMapper();
+    String programConfigPath;
+    JSONObject jsonContent;
+
+    try {
+      JsonNode params = mapper.readTree(pTransform.getSpec().getPayload().toByteArray());
+      Preconditions.checkNotNull(
+          programConfigPath = params.path(
+              "program_config_path").textValue(),
+          "'program_config_path' needs to be set");
+
+
+      File programConfigFile = new File(programConfigPath);
+      if (!programConfigFile.exists()) {
+        throw new IllegalArgumentException("File not found at " + programConfigPath);
+      }
+
+      InputStream inputStream = new FileInputStream(programConfigPath);
```

Comment:
We should probably move everything that does not strictly belong to the translation into the Dryft SDK.

Reply:
Done! Done. https://github.com/lyft/streamingplatform/pull/1572